File: test-framework/src/main/java/api/app/astrodao/com/core/dto/api/proposals/bounty/BountyDto.java
package api.app.astrodao.com.core.dto.api.proposals.bounty;

import api.app.astrodao.com.core.dto.api.proposals.Dao;
import com.fasterxml.jackson.annotation.JsonProperty;

public class BountyDto {
    @JsonProperty("createdAt")
    public String createdAt;
    @JsonProperty("transactionHash")
    public Object transactionHash;
    @JsonProperty("updateTransactionHash")
    public Object updateTransactionHash;
    @JsonProperty("createTimestamp")
    public Object createTimestamp;
    @JsonProperty("updateTimestamp")
    public Object updateTimestamp;
    @JsonProperty("id")
    public String id;
    @JsonProperty("bountyId")
    public Integer bountyId;
    @JsonProperty("daoId")
    public String daoId;
    @JsonProperty("description")
    public String description;
    @JsonProperty("token")
    public String token;
    @JsonProperty("amount")
    public String amount;
    @JsonProperty("times")
    public String times;
    @JsonProperty("maxDeadline")
    public String maxDeadline;
    @JsonProperty("numberOfClaims")
    public Integer numberOfClaims;
    @JsonProperty("dao")
    public Dao dao;
}

File: libs/bounty/src/dto/bounty.dto.ts
import { TransactionInfo } from '@sputnik-v2/common';
import { BountyClaimDto } from './bounty-claim.dto';

export class BountyDto extends TransactionInfo {
  id: string;
  daoId: string;
  bountyId: number;
  proposalId: string;
  proposalIndex: number;
  description: string;
  /// Token the bounty will be paid out.
  token: string;
  /// Amount to be paid out.
  amount: string;
  /// How many times this bounty can be done.
  times: number;
  /// Max deadline from claim that can be spend on this bounty.
  maxDeadline: string;
  numberOfClaims: number;
  bountyClaims: BountyClaimDto[];
}
